services/navigation/builtin/builtin.go

```diff
@@ -412,17 +442,21 @@ func (svc *builtIn) startWaypoint(ctx context.Context, extra map[string]interfac
 					svc.logger.Infof("skipping waypoint %+v since it was deleted", wp)
 					continue
 				}
+				svc.logger.Warnf("failed to navigate to waypoint: %+v due to error %s", wp, err.Error())
+				continue
+			}
+
+			// mark the waypoint as reached if the motion service hit no error
+			// TODO: We probably need to terminate if we can't reach the mongo DB
 
-				svc.logger.Infof("skipping waypoint %+v due to error while navigating towards it: %s", wp, err)
-				if err := svc.waypointReached(ctx); err != nil {
-					if svc.waypointIsDeleted() {
-						svc.logger.Infof("skipping waypoint %+v since it was deleted", wp)
-						continue
-					}
-					svc.logger.Infof("can't mark waypoint %+v as reached, exiting navigation due to error: %s", wp, err)
-					return
+			if err = svc.waypointReached(ctx); err != nil {
+				svc.logger.Errorf("failed to update the waypoint db %+v, calling Close() on nav", wp)
+				if err = svc.Close(ctx); err != nil {
+					svc.logger.Errorf("navigation Close() returned an error", err)
```

Comment:
If we fail to mark the waypoint as reached in the store (which only happens if we are using the mongo store & mongo is unavailable) we close the nav service as not being able to record waypoints you have reached is an irrecoverable error.